release-22.2: vendor: bump Pebble to 20de6220b828

jbowens opened the pull request:
```
20de6220 db: truncate range tombstones to containing file's atomic unit
```

Release note: None
Release justification: Fixes a high severity data loss issue.
Epic: None
Informs #89777.

nicktrav reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/90215#-:-NEhmXyzDRQmYGX_DF4p:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jbowens)



<!-- Sent from Reviewable.io -->

jbowens committed:
vendor: bump Pebble to 20de6220b828

20de6220 db: truncate range tombstones to containing file's atomic unit

Release note: None

jbowens commented:
tftr!
